pymeasure/instruments/thorlabs/thorlabspm100usb.py

```
@@ -57,74 +62,168 @@ def wavelength(self):
-            self.is_energy_sensor,  # 2
-            _d4,  # 4
-            _d8,  # 8
-            self.response_settable,  # 16
-            self.wavelength_settable,  # 32
-            self.tau_settable,  # 64
-            _d128,  # 128
-            self.has_temperature_sensor,  # 256
-        ) = self.flags
+        self.sensor_name = response[0].replace('"', "")
+        self.sensor_sn = (
+            response[1].strip('"') if type(response[1]) == str else str(int(response[1]))
```

Ruff (0.15.12)

[error] Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)